runtime/src/bank.rs

```diff
@@ -1675,8 +1675,20 @@ impl Bank {
     }
 
     /// Return subset of bank fields representing serializable state
-    pub(crate) fn get_fields_to_serialize(&self) -> BankFieldsToSerialize {
-        let (epoch_stakes, versioned_epoch_stakes) = split_epoch_stakes(self.epoch_stakes.clone());
+    pub(crate) fn get_fields_to_serialize(
+        &self,
+        split_epoch_stakes_enabled: bool,
+    ) -> BankFieldsToSerialize {
+        // The `split_epoch_stakes_enabled` arg only exists to allow serializing
+        // a bank without any of the extra fields in tests to simulate creating
+        // old snapshots from previous releases. It should never be `false`
+        // under normal operations.
```

Comment:
If this assertion is true, I would prefer we add a separate test-only method to handle the `false` case.

Reply:
Removed this arg